### PR TITLE
unicode fixes

### PR DESCRIPTION
**Pull request by siiptuo**

### CODE REVIEW

**Comment by Cloudef** [comment]
Can you describe what the problem here is?

File: lib/renderers/cairo.h [hunk 1]
```diff
@@ -155,11 +155,8 @@ bm_cairo_draw_line(struct cairo *cairo, struct cairo_paint *paint, struct cairo_
     pango_cairo_show_layout(cairo->cr, layout);
 
     if (paint->draw_cursor) {
-        size_t chr = 0;
-        for (size_t c = 0; c < paint->cursor; ++chr, c += bm_utf8_rune_next(buffer, c));
-
         PangoRectangle rect;
-        pango_layout_index_to_pos(layout, chr, &rect);
+        pango_layout_index_to_pos(layout, paint->cursor, &rect);
```
Comment (Cloudef) [inline lib/renderers/cairo.h:160]: This most likely will break multibyte text. (CJK)

Comment (Cloudef) [reply lib/renderers/cairo.h:160]: Note that chr is byte index.

File: lib/menu.c [hunk 2]
```diff
@@ -710,8 +710,11 @@ bm_menu_run_with_key(struct bm_menu *menu, enum bm_key key, uint32_t unicode)
             break;
 
         case BM_KEY_DELETE:
-            if (menu->filter)
-                bm_utf8_rune_remove(menu->filter, menu->cursor + 1, NULL);
+            if (menu->filter) {
+                size_t width = bm_utf8_rune_next(menu->filter, menu->cursor);
+                if (width)
+                    bm_utf8_rune_remove(menu->filter, menu->cursor + width, NULL);
+            }
```
Comment (Cloudef) [inline lib/menu.c:717]: This looks correct :+1:

File: lib/renderers/cairo.h (hunk 1 above)
Comment (Cloudef) [reply lib/renderers/cairo.h:160]: Or isn't, I can't read. Weird, are you sure this takes byte index? The documentation seems to talk about "grapheme".
>Converts from an index within a PangoLayout to the onscreen position corresponding to the grapheme at that index

Comment (Cloudef) [reply lib/renderers/cairo.h:160]: >index_ | byte index within layout

Seems so.. I remember this not working with simply `paint->cursor` though :S
Can you test with some CJK text?

**Comment by siiptuo** [comment]
> Can you describe what the problem here is?

Trying to fix issues I noticed after typing something like "äää" and moving the cursor around.

File: lib/renderers/cairo.h (hunk 1 above)
Comment (siiptuo) [reply lib/renderers/cairo.h:160]: CJK text seems to work. I don't have a good way of typing these characters so I tested with `bm_menu_set_filter` and `xdotool type`.

**Comment by Cloudef** [comment]
Okay, merging this. The changes seem to make sense.

**Comment by Cloudef** [comment]
Thanks!

**Pull request merged**